client/src/components/MultiJurisdictionAudit/Progress/ProgressMap.tsx

```diff
@@ -0,0 +1,353 @@
+import React, { useState, useRef, useEffect, useCallback } from 'react'
+import styled from 'styled-components'
+import { select, json, geoPath, geoAlbers } from 'd3'
+import { feature } from 'topojson-client'
+// topojson-specification is defined in package.json but throwing linting error here
+// eslint-disable-next-line import/no-unresolved
+import { Topology } from 'topojson-specification'
+import { Colors, Spinner } from '@blueprintjs/core'
+import labelValueStates from '../AASetup/Settings/states'
+import {
+  IJurisdiction,
+  getJurisdictionStatus,
+  JurisdictionProgressStatus,
+} from '../useJurisdictions'
+import { IAuditSettings } from '../useAuditSettings'
+
+interface IProps {
+  stateName: string | null
+  jurisdictions: IJurisdiction[]
+  isRoundStarted: boolean
+  auditType: IAuditSettings['auditType']
+}
+
+const MapWrapper = styled.div`
+  position: relative;
+`
+
+const SVGMap = styled.svg`
+  margin-bottom: 30px;
+  .outline {
+    fill: none;
+    stroke: ${Colors.BLACK};
+    stroke-width: 0.5px;
+  }
+  .mesh {
+    fill: none;
+    stroke-linejoin: round;
+  }
+  path {
+    fill: none;
+    stroke: none;
+    stroke-width: 0.5px;
+  }
+  #single-state {
+    stroke: ${Colors.BLACK};
+  }
+  .county {
+    fill: ${Colors.WHITE};
+    stroke: ${Colors.BLACK};
+  }
+  .county:hover {
+    stroke-width: 2px;
+  }
+  .danger {
+    fill: ${Colors.RED3};
+  }
+  .success {
+    fill: ${Colors.GREEN4};
+  }
+  .progress {
+    fill: ${Colors.ORANGE4};
+  }
+  .gray {
+    fill: ${Colors.GRAY4};
+  }
+  .default {
+    fill: ${Colors.WHITE};
+  }
+`
+
+const Tooltip = styled.div`
+  display: none;
+  position: absolute;
+  opacity: 0.9;
+  border-radius: 3px;
+  background: ${Colors.BLACK};
+  padding: 5px;
+  color: ${Colors.WHITE};
+`
+
+const MapLabels = styled.div`
+  position: absolute;
+  top: 0;
+  z-index: 10;
+`
+
+const MapLabelsRow = styled.div`
+  display: flex;
+  align-items: center;
+  margin-bottom: 5px;
+`
+
+const MapLabelsBoxes = styled.div`
+  display: inline-block;
+  margin-right: 10px;
+  width: 20px;
+  height: 20px;
+  &.danger {
+    background-color: ${Colors.RED3};
+  }
+  &.success {
+    background-color: ${Colors.GREEN4};
+  }
+  &.progress {
+    background-color: ${Colors.ORANGE4};
+  }
+  &.gray {
+    background-color: ${Colors.GRAY4};
+  }
+  &.default {
+    border: 1px solid ${Colors.BLACK};
+  }
+`
+
+const MapSpinner = styled(Spinner)`
+  position: absolute;
+  top: 50%;
+  right: 0;
+  left: 0;
+  transform: translateY(-50%);
+  margin: 0 auto;
+`
+
+const Map = ({
+  stateName,
+  jurisdictions,
+  isRoundStarted,
+  auditType,
+}: IProps) => {
+  const width = 960
+  const height = 500
+  const d3Container = useRef(null)
+  const tooltipContainer = useRef(null)
+
+  const [jsonData, setJsonData] = useState<Topology | undefined>(undefined)
+  const [matchedJurisdictions, setMatchedJurisdictions] = useState<boolean>(
```

Comment:
i think this is derived from `jsonData` and `jurisdictions`, so it doesn't need  to be its own state variable (less state => less confusing state bugs 😄 )